drivers/input/input_npcx_kbd.c

```diff
@@ -453,10 +176,15 @@ static int input_npcx_kbd_init(const struct device *dev)
 	}
 
 	/* Drive all column lines to low for detection any key press */
-	input_npcx_kbd_drive_column(dev, KEYBOARD_COLUMN_DRIVE_NONE);
+	input_npcx_kbd_drive_column(dev, INPUT_KBD_MATRIX_COLUMN_DRIVE_NONE);
+
+	if (common->row_size != ROW_SIZE) {
```

Comment:
Non-blocking - what are you guarding against here? I'm not sure how this would ever fail.

Reply:
The fact that `ksi_callback` is statically allocated on `ROW_SIZE` but iterated on `common->row_size`, it should indeed never fail but the fact that these two are set in different files jumping through macros all over the place made me think it's a good idea to bail out if for whatever reasons they end up diverging.

Not too attached to it though, lmk if you want me to drop it, or maybe I could do a `INPUT_KBD_MATRIX_DT_INST_ROW_SIZE` and use that to size `ksi_callback`.